sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -144,6 +199,51 @@ public static IDisposable CreateHttpMessagePropertiesScope(IDictionary<string, o
             return CurrentHttpMessagePropertiesScope.Value;
         }
 
+        private ReadOnlyMemory<HttpPipelinePolicy> CreateRequestPipeline(HttpPipelinePolicy[] policies, List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)> customPolicies)
+        {
+            if (!_internallyConstructed)
+            {
+                throw new InvalidOperationException("Cannot send messages with per-request policies if the pipeline wasn't constructed with HttpPipelineBuilder.");
+            }
+
+            int AddCustomPolicies(HttpPipelinePosition position, int start)
+            {
+                int i = 0;
+                if (customPolicies != null)
```

Comment:
`customPolicies` should never be null on this pass, so I think we can just throw